src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -918,4 +919,14 @@ public boolean test(SCMTraitDescriptor<?> scmTraitDescriptor) {
         }
     }
 
+    @SuppressFBWarnings(value="RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification="Known to be null during deserialization")
```

Comment:
```suggestion
```

I suggest you move the initialization to the constructor to avoid the need for `SuppressFBWarnings`
Honestly thought that they would be initialized when on the field 😭

Reply:
Just tried with no success:
```diff
diff --git a/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java b/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java
index 695492b..1423473 100644
--- a/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java
+++ b/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java
@@ -119,18 +119,20 @@
      * The cache of {@link ObjectMetadataAction} instances for each open MR.
      */
     @NonNull
-    private transient /*effectively final*/ Map<Integer,ObjectMetadataAction> mergeRequestMetadataCache = new ConcurrentHashMap<>();
+    private transient /*effectively final*/ Map<Integer,ObjectMetadataAction> mergeRequestMetadataCache;
     /**
      * The cache of {@link ObjectMetadataAction} instances for each open MR.
      */
     @NonNull
-    private transient /*effectively final*/ Map<Integer,ContributorMetadataAction> mergeRequestContributorCache = new ConcurrentHashMap<>();
+    private transient /*effectively final*/ Map<Integer,ContributorMetadataAction> mergeRequestContributorCache;
 
     @DataBoundConstructor
     public GitLabSCMSource(String serverName, String projectOwner, String projectPath) {
         this.serverName = serverName;
         this.projectOwner = projectOwner;
         this.projectPath = projectPath;
+        mergeRequestMetadataCache = new ConcurrentHashMap<>();
+        mergeRequestContributorCache = new ConcurrentHashMap<>();
     }
 
     public String getServerName() {
@@ -918,15 +920,4 @@
             };
         }
     }
-
-    @SuppressFBWarnings(value="RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification="Known to be null during deserialization")
-    private Object readResolve() {
-        if (mergeRequestMetadataCache == null) {
-            mergeRequestMetadataCache = new ConcurrentHashMap<>();
-        }
-        if (mergeRequestContributorCache == null) {
-            mergeRequestContributorCache = new ConcurrentHashMap<>();
-        }
-        return this;
-    }
 }
```

```
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at org.junit.Assert.assertNotNull(Assert.java:722)
	at io.jenkins.plugins.gitlabbranchsource.GitLabSCMSourceDeserializationTest.afterDeserializationWithXStreamTransientFieldsAreNotNull(GitLabSCMSourceDeserializationTest.java:19)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)


```

Reply:
I meant you should keep the `readResolve`, but remove the `SuppressFBWarnings` :)

Reply:
Ah, now I see the idea. Just put initialization into constructor and removed suppression -- it did not pass the validation: 
```
D:/project/git/gitlab-branch-source-plugin/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java:924 Redundant nullcheck of io.jenkins.plugins.gitlabbranchsource.GitLabSCMSource.mergeRequestMetadataCache, which is known to be non-null in io.jenkins.plugins.gitlabbranchsource.GitLabSCMSource.readResolve() [Of Concern(18), Normal confidence]
D:/project/git/gitlab-branch-source-plugin/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java:927 Redundant nullcheck of io.jenkins.plugins.gitlabbranchsource.GitLabSCMSource.mergeRequestContributorCache, which is known to be non-null in io.jenkins.plugins.gitlabbranchsource.GitLabSCMSource.readResolve() [Of Concern(18), Normal confidence]
```

Reply:
Somewhat sad state of affairs 😭